### ISSUE TITLE
Parse error

### ISSUE THREAD

**Issue by taviroquai** [creof/wkb-parser#13]
Hi,

I'm getting a parse error: Unsupported WKB type "1001".

This is the input:
$parser->parse('00000003e94117c89f84189375411014361ba5e3540000000000000000');

This input is POINT(389671.879 263437.527 0)

I could see because it was parsed ok by the the javascript Node module:
https://github.com/cschwarz/wkx

But I need to process this on the server side :)

I think it has something to do with the byte order.

Thanks!

**Comment by djlambert** [comment]
Are you passing the WKB value as a string like you have shown?

Currently the Parser/Reader expects the value to be a binary string or a text hexadecimal string beginning with `'x'` or `'0x'``. So you could do:

`pack('H*', '00000003e94117c89f84189375411014361ba5e3540000000000000000')`

or prepend `'0x'` to the beginning of the string.

I'll update the documentation with this. I'll take a look at the detection logic too for possible enhancement.

**Comment by taviroquai** [comment]
Thanks for the info :)

Strange I always get the same error, even when I try both of your indications...

Can you try to reproduce the error with that input? Thanks :)

**Comment by djlambert** [comment]
My previous comment was incorrect. The Reader will in fact take straight hex with no prefix. 

Manually parsing your string the object type is indeed 1001 (0x000003e9).

It appears wkx uses a different set of flags in generated WKB to indicate 3D and 4D geometry when an object has an [undefined SRID](https://github.com/cschwarz/wkx/blob/master/lib/geometry.js#L305). Currently this code handles the flags used by PostGIS EWKB, and I'd presume these other flags are used by another database platform. I'd love to add support for it if I can find documentation on this. What database are you using?

**Comment by djlambert** [comment]
I've opened up cschwarz/wkx#11 to hopefully get additional information.

**Comment by djlambert** [comment]
Looks like OGC added 3D and 4D support to the standard in 2006, and the flags used by PostGIS differ. I'll work on incorporating these.

**Comment by taviroquai** [comment]
Cool I got your interest :)

So this is the story...

I'm working on the project mapigniter2 https://github.com/taviroquai/mapigniter2 and I'm adding support to GeoPackage standard which you can see the spec in http://www.geopackage.org/spec/

They introduced an header which contains some flags, an envelope and the E/WKB geomety.
This is the relevant part about the header http://www.geopackage.org/spec/#gpb_format

To test my code, I'm using the sample data from http://www.geopackage.org/data/simple_sewer_features.gpkg
This file is a new version of SQLite and I'm getting the geometries from the s_manhole table.
This geometry is a BLOB defined by the spec in geopackage.org

You can see my current parser in my code in here:
https://github.com/taviroquai/mapigniter2/blob/master/app/GeoPackage.php#L220

As we can see that the wkx javascript library parses fine the extracted WKB from the GeoPackage format, maybe the fix is something simple but I can't see where...

I hope this will be easy for you! :)

**Comment by djlambert** [comment]
Think I've got the code worked out, just need to finish the tests. Should have something in the next day or two.

**Comment by djlambert** [comment]
The code will now properly handle the OGC types, you can try the develop branch of my fork djlambert/wkb-parser if you want. Have some cleanup and tweaks to do before pulling it over here. I like how you're using fread in your package, it's a lot cleaner than what I'm currently doing.

**Comment by taviroquai** [comment]
Hi Derek,

As you said, I'm trying with your develop branch and it's working :)

I also add to convert to GeoJSON object so I had to add the following code:

```
$wkbNames = ['POINT', 'LINESTRING', 'POLYGON'];
$geojsonNames = ['Point', 'LineString', 'Polygon'];
$feature['geometry'] = $parser->parse($wkb);
$feature['geometry']['type'] = str_replace($wkbNames, $geojsonNames, $feature['geometry']['type']);
$feature['geometry']['coordinates'] = $feature['geometry']['value'];
unset($feature['geometry']['srid']);
unset($feature['geometry']['value']);
```

I'm going to create another issue to request geometry GeoJSON output object.

Good work! :)

**Comment by taviroquai** [comment]
Hi Derek,

When the support for OGC v1.2 will be available on a release, and which release?

Thanks! :)

**Comment by djlambert** [comment]
It'll be 2.3, I have #19 open for it. Going to update the date and merge it.

**Comment by djlambert** [comment]
Support for OGC v1.2 encoding is now available in release v2.3.
